public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -554,89 +552,77 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
+    const splitKeys = tagKeys.split(',');
 
-      if (results.cancelled) {
-        return [];
-      }
-      _.each(results.data.data.result, series => {
-        const tags = _.chain(series.metric)
-          .filter((v, k) => {
-            return _.includes(tagKeys, k);
-          })
-          .value();
-
-        const dupCheck: { [key: number]: boolean } = {};
-        for (const value of series.values) {
-          const valueIsTrue = value[1] === '1'; // e.g. ALERTS
-          if (valueIsTrue || annotation.useValueForTime) {
-            const event: AnnotationEvent = {
-              annotation: annotation,
-              title: self.resultTransformer.renderTemplate(titleFormat, series.metric),
-              tags: tags,
-              text: self.resultTransformer.renderTemplate(textFormat, series.metric),
-            };
-
-            if (annotation.useValueForTime) {
-              const timestampValue = Math.floor(parseFloat(value[1]));
-              if (dupCheck[timestampValue]) {
-                continue;
-              }
-              dupCheck[timestampValue] = true;
-              event.time = timestampValue;
-            } else {
-              event.time = Math.floor(parseFloat(value[0])) * 1000;
-            }
+    if (response.cancelled) {
+      return [];
+    }
 
-            eventList.push(event);
+    response?.data?.data?.result?.forEach(series => {
+      const tags = Object.entries(series.metric)
+        .filter(([k]) => splitKeys.includes(k))
+        .map(([_k, v]: [string, string]) => v);
+
+      const dupCheck: Record<number, boolean> = {};
+      for (const value of series.values) {
+        const valueIsTrue = value[1] === '1'; // e.g. ALERTS
+        if (valueIsTrue || annotation.useValueForTime) {
+          const event: AnnotationEvent = {
+            annotation,
+            title: self.resultTransformer.renderTemplate(titleFormat, series.metric),
+            tags,
+            text: self.resultTransformer.renderTemplate(textFormat, series.metric),
+          };
+
+          if (annotation.useValueForTime) {
+            const timestampValue = Math.floor(parseFloat(value[1]));
+            if (dupCheck[timestampValue]) {
+              continue;
+            }
+            dupCheck[timestampValue] = true;
+            event.time = timestampValue;
+          } else {
+            event.time = Math.floor(parseFloat(value[0])) * 1000;
           }
-        }
-      });
 
-      return eventList;
+          eventList.push(event);
+        }
+      }
     });
+
+    return eventList;
   }
 
-  getTagKeys(options: any = {}) {
-    const url = '/api/v1/labels';
-    return this.metadataRequest(url).then((result: any) => {
-      return _.map(result.data.data, value => {
-        return { text: value };
-      });
-    });
+  async getTagKeys() {
+    const result = await this.metadataRequest('/api/v1/labels');
+    return result?.data?.data?.map((value: any) => ({ text: value }));
```

Comment:
@kaydelaney hmm I just realised that the lodash map returns an empty array if the value is null/undefined but this will return undefined/null in that case essentially changing fucntion return type from `T[]` to `T[] | undefined | null`. Is this Ok in this context, not sure if we test for that case.

Reply:
added a fallback to follow the old behavior.